scripts/loadDbSettings.ts

```diff
@@ -0,0 +1,85 @@
+import { join } from "path";
+import { readFile } from "fs/promises";
+import pgp, { IDatabase } from "pg-promise";
+
+const pgPromiseLib = pgp();
+
+type Database = IDatabase<{}>;
+
+const modes = ["dev", "prod", "staging", "testing", "xpost-dev"] as const;
+
+type Mode = typeof modes[number];
+
+type Settings = {
+  databasePublicSettings: Record<string, any>;
+  databaseServerSettings: Record<string, any>;
+}
+
+const isMode = (value: string): value is Mode =>
+  modes.includes(value as Mode);
+
+const capitalize = (s: string) => s[0].toUpperCase() + s.slice(1);
+
+const makeCredentialsPath = (fileName: string) =>
+  join("..", "..", "ForumCredentials", fileName);
+
+const readSettingsFile = (mode: Mode): Settings => {
+  if (mode === "xpost-dev") {
+    mode = "dev";
+  }
+  const path = makeCredentialsPath(`load${capitalize(mode)}DbSettings.js`);
+  const data = require(path);
```

Comment:
You know, require is basically an eval, isn't it.

Reply:
Yeah - _marginally_ less janky